soil/__init__.py

```diff
@@ -142,6 +143,9 @@ def set_progress(cls, task, current, total):
                 task.update_state(state='PROGRESS', meta={'current': current, 'total': total})
         except (TypeError, NotImplementedError):
             pass
+        except IntegrityError:
+            # Not called in task context just pass
+            pass
```

Comment:
What about a check for `task.request.id`?

http://celery.readthedocs.org/en/latest/_modules/celery/app/task.html#Task.update_state